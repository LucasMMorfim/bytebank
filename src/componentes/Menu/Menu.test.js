import { render, screen } from '@testing-library/react';
import Menu from './index';

it('Render a link to the initial page', () => {
  render(<Menu />);
  const linkInicialPage = screen.getByText('Inicial');
  expect(linkInicialPage).toBeInTheDocument();
});

it('Render a list of links', () => {
  render(<Menu />);
  const listOfLinks = screen.getAllByRole('link');
  expect(listOfLinks).toHaveLength(4);
});

it('Should be not render an link of extract', () => {
  render(<Menu />);
  const extractLink = screen.queryByText('Extrato');
  expect(extractLink).not.toBeInTheDocument();
});

it('Should be render a list of links with link class', () => {
  render(<Menu />);
  const links = screen.getAllByRole('link');
  links.forEach((link) => expect(link).toHaveClass('link'))
  expect(links).toMatchSnapshot()
})