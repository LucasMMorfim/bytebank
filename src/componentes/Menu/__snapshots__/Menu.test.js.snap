// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should  render a list of links with link class 1`] = `
Array [
  <a
    class="link"
    href="/"
  >
    Inicial
  </a>,
  <a
    class="link"
    href="/"
  >
    Transferências
  </a>,
  <a
    class="link"
    href="/"
  >
    Investimentos
  </a>,
  <a
    class="link"
    href="/"
  >
    Outros serviços
  </a>,
]
`;

exports[`Should be render a list of links with link class 1`] = `
Array [
  <a
    class="link"
    href="/"
  >
    Inicial
  </a>,
  <a
    class="link"
    href="/"
  >
    Transferências
  </a>,
  <a
    class="link"
    href="/"
  >
    Investimentos
  </a>,
  <a
    class="link"
    href="/"
  >
    Outros serviços
  </a>,
]
`;

exports[`Should render a list of links with link class 1`] = `
Array [
  <a
    class="link"
    href="/"
  >
    Inicial
  </a>,
  <a
    class="link"
    href="/"
  >
    Transferências
  </a>,
  <a
    class="link"
    href="/"
  >
    Investimentos
  </a>,
  <a
    class="link"
    href="/"
  >
    Outros serviços
  </a>,
]
`;
